protocol/x/vest/keeper/keeper.go

```diff
@@ -120,7 +121,27 @@ func (k Keeper) ProcessVesting(ctx sdk.Context) {
 				entry.TreasuryAccount,
 				sdk.NewCoins(sdk.NewCoin(entry.Denom, vestAmount)),
 			); err != nil {
-				panic(err)
+				// This should never happen. However, if it does, we should not panic.
```

Comment:
Thanks for making this change. I explicitly wanted to panic instead of logging an error during implementation, since this is an invariant violation. 

However after second thought, I'm inclined to agree that logging an error makes more sense here. This is because failure to process a vest event does not disrupt core trading flows. It will not lead to protocol/user losing funds (although I can lead to user not getting their entitled rewards). 

I started a thread in protocol to be aligned on similar topic. Will approve after that thread is resolved.